vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java

```diff
@@ -150,4 +150,89 @@ public void whenRefererMatchesAControllerShouldForwardToIt() throws Exception {
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.redirectTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.redirect();
+		
+		verify(logic).redirectTo(RefererController.class);
+		verify(controller).index();
+	}
+
+	@Test
+	public void whenRefererContainsCtxPathStrInTheHostItShouldForwardCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/vrap/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.forwardTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.forward();
+		
+		verify(logic).forwardTo(RefererController.class);
+		verify(controller).index();
+	}
+	
+	@Test
+	public void whenRefererDoesNotContainsCtxPathItShouldRedirectCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.redirectTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.redirect();
+		
+		verify(logic).redirectTo(RefererController.class);
+		verify(controller).index();
+	}
+	
+	@Test
+	public void whenRefererDoesNotContainsCtxPathItShouldForwardCorrectly() throws Exception {
+		LogicResult logic = mock(LogicResult.class);
+		RefererController controller = mock(RefererController.class);
+		
+		Method index = RefererController.class.getMethod("index");
+		ControllerMethod method = DefaultControllerMethod.instanceFor(RefererController.class, index);
+		
+		when(request.getHeader("Referer")).thenReturn("http://vraptor.test.com/anything/ok");
+		when(request.getContextPath()).thenReturn("/vrap");
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.forwardTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.forward();
+		
+		verify(logic).forwardTo(RefererController.class);
+		verify(controller).index();
+	}
```

Comment:
You're right, I followed the other methods pattern but we could have simple tests which would check if the returned string matches the expected value and we're good to go. Wanna me to change that?